### PR TITLE
#466. Deregister Conversation mostly implemented.

### PR DESCRIPTION
**Pull request by criske**
- implemented and tested AuthorHasRoles step.
- implemented Deregister conversation.
- mostly implemented RemoveTask step.
- left puzzle to test Deregister,RemoveTask + finish it.
- left puzzle to remove a task from db.

PR for #466

### CODE REVIEW

File: self-core-impl/src/main/java/com/selfxdsd/core/managers/AuthorHasRoles.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @Override
+    public void perform(final Event event) {
+        final Project project = event.project();
+        final String author = event.comment().author();
+        final Contributor contributor = project
+            .contributors()
+            .getById(author, project.provider());
+        if (contributor == null) {
```
Comment (criske) [inline self-core-impl/src/main/java/com/selfxdsd/core/managers/AuthorHasRoles.java:76]: is `project.owner()` implicit a contributor? 
Else I should check to see if author is project.owner() and has the role.

Comment (amihaiemil) [reply self-core-impl/src/main/java/com/selfxdsd/core/managers/AuthorHasRoles.java:76]: @criske The owner is not by default a contributor. I think we will make him a Contributor with role ``PO`` (Project Owner), but it's irrelevant for the code here.

When we register a Project, we could call ``Contributors.register(...)`` to add him as a contributor. So don't worry.

File: self-core-impl/src/main/java/com/selfxdsd/core/managers/Deregister.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import com.selfxdsd.api.pm.Conversation;
+import com.selfxdsd.api.pm.Step;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Conversation where a PM will unassign the Task and remove it from DB.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.20g
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/managers/Deregister.java:37]: @criske typo here, can you remove the ``g``?

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**